documentation/scaling_puppet_server.md

```diff
@@ -0,0 +1,131 @@
+
+```
+[main]
+use_srv_records = true
+srv_domain = example.com
+```
+
+Agents will then lookup a `SRV` record at `_x-puppet._tcp.example.com` when they need to talk to a Puppet master.
+
+```
+# Equal-weight load balancing between master-a and master-b:
+_x-puppet._tcp.example.com. IN SRV 0 5 8140 master-a.example.com.
+_x-puppet._tcp.example.com. IN SRV 0 5 8140 master-b.example.com.
+```
+
+You can also implement more complex configurations. For instance, if all devices in site A are configured with a `srv_domain` of `site-a.example.com`, and all nodes in site B are configured to `site-b.example.com`, you can configure them to prefer a master in the local site but fail over to the remote site:
+
+```
+# Site A has two masters - master-1 is beefier, give it 75% of the load:
+_x-puppet._tcp.site-a.example.com. IN SRV 0 75 8140 master-1.site-a.example.com.
+_x-puppet._tcp.site-a.example.com. IN SRV 0 25 8140 master-2.site-a.example.com.
+_x-puppet._tcp.site-a.example.com. IN SRV 1 5 8140 master.site-b.example.com.
+
+# For site B, prefer the local master unless it's down, then fail back to site A
+_x-puppet._tcp.site-b.example.com. IN SRV 0 5 8140 master.site-b.example.com.
+_x-puppet._tcp.site-b.example.com. IN SRV 1 75 8140 master-1.site-a.example.com.
+_x-puppet._tcp.site-b.example.com. IN SRV 1 25 8140 master-2.site-a.example.com.
+```
+
+## Centralizing the Certificate Authority
+
+Additional Puppet Servers should only share the burden of compiling and serving catalogs, which is why they're typically referred to as "compile masters". Any certificate authority functions should be delegated to a single server.
+
+Before you centralize this functionality, ensure that the single server that you want to use as the central CA is reachable at a unique hostname other than (or in addition to) `puppet`. Next, point all agent requests to the centralized CA master, either by configuring each agent or through DNS `SRV` records.
+
+### Directing individual agents to a central CA
+
+On every agent, set the [`ca_server`](https://docs.puppet.com/puppet/latest/reference/configuration.html#caserver) setting in [`puppet.conf`](https://docs.puppet.com/puppet/latest/reference/config_file_main.html) (in the `[main]` configuration block) to the hostname of the server acting as the certificate authority. If you have a large number of existing nodes, it is easiest to do this by managing `puppet.conf` with a Puppet module and a template.
+
+> **Note:** Set this setting *before* provisioning new nodes, or they won't be able to complete their initial agent run.
+
+### Pointing DNS `SRV` records at a central CA
+
+If you [use `SRV` records for agents](#using-dns-srv-records), you can use the `_x-puppet-ca._tcp.$srv_domain` DNS name to point clients to one specific CA server, while the `_x-puppet._tcp.$srv_domain` DNS name handles most of their requests to masters and can point to a set of compile masters.
+
```

Comment:
Similarly, PE users will want to avoid tinkering with the CA settings as the `puppet_enterprise` module handles this.